dynamo_client.py

```diff
@@ -23,47 +21,61 @@ def get_customers(self, source_customer_id):
         
     def put_customer(self, customer):
         response = self.cust_table.put_item(Item={
-            'customer_id':customer.customer_id,
-            'source_name':customer.source_name,
-            'source_customer_id':customer.source_customer_id,
-            'orders':customer.orders,
+            'customer_id': customer.customer_id,
+            'source_name': customer.source_name,
+            'source_customer_id': customer.source_customer_id,
+            'orders': customer.orders,
             'upserted_at': customer.upserted_at
         })
 
         return response['ResponseMetadata']['HTTPStatusCode'] == 200
     
     def put_order(self, order, customer):
         response = self.order_table.put_item(Item={
-                    'order_id':order.id,
-                    'source_name':order.source_name,
-                    'source_order_id': order.source_order_id,
-                    'customer_id': customer.customer_id,
-                    'total_price': Decimal(str(order.price)),
-                    'order_items': order.order_items,
-                    'upserted_at': str(datetime.now())
-                })
+            'order_id': order.id,
+            'source_name': order.source_name,
+            'source_order_id': order.source_order_id,
+            'customer_id': customer.customer_id,
+            'total_price': Decimal(str(order.price)),
+            'order_items': order.order_items,
+            'upserted_at': str(datetime.now())
+        })
         
         return response['ResponseMetadata']['HTTPStatusCode'] == 200
     
     def put_esim_order(self, esim_order, order_table_ref):
         esim_order = json.loads(esim_order)
-        print(type(esim_order))
+        # Convert 'total' field from float to Decimal
+        esim_order['total'] = Decimal(str(esim_order['total']))
+        for order_item in esim_order['order']:
+            print(order_item)
+            order_item['subTotal'] = Decimal(str(order_item['subTotal']))
+            order_item['pricePerUnit'] = Decimal(str(order_item['pricePerUnit']))
+        
         response = self.esim_table.put_item(Item={
-                    'esim_order_id':esim_order['orderReference'],
-                    'order_table_ref_id':order_table_ref.id,
-                    'status':esim_order['status'],
-                    'currency': esim_order['currency'],
-                    'total': esim_order['total'],
-                    'order':esim_order['order'],
-                    'upserted_at': str(datetime.now())
-                })
+            'esim_order_id': esim_order['orderReference'],
+            'order_table_ref_id': order_table_ref.id,
+            'status': esim_order['status'],
+            'currency': esim_order['currency'],
+            'total': esim_order['total'],
+            'order': esim_order['order'],
+            'upserted_at': str(datetime.now())
```

Comment:
just checking that these dates are UTC or derived from the locality of the lambda?

Reply:
current date and time in the timezone of the Lambda.

Reply:
@jamweee do we need utc?

Reply:
@zaine2003 changes done -  upserted_at time to UTC